### PR TITLE
Add .gitlab-ci.yml

### PR DESCRIPTION
**Pull request by jgmize**
## Description

Gitlab build & deploy jobs for stage & prod for mozmeao/infra#985

- [x] record this change in `CHANGELOG.md`
- [x] follow steps documented in https://github.com/mozmeao/infra/blob/master/docs/gitlab_ci.md to finish gitlab integrations

### Issue

mozmeao/infra#985

### Testing

Successful stage pipeline: https://gitlab.com/mozmeao/protocol/-/jobs/378292594

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 10.0.2
```
Comment (pmac) [inline CHANGELOG.md]: Aren't these version numbers for npm releases?

Comment (jgmize) [reply CHANGELOG.md]: Happy to remove this, thanks!

**Comment by jgmize** [comment]
Rebased and squashed, omitting the CHANGELOG.md commit.

File: CHANGELOG.md (hunk 1 above)
Comment (pmac) [reply CHANGELOG.md]: To be clear, I'm not sure. Let's wait for @stephaniehobson to weigh-in.

Comment (craigcook) [reply CHANGELOG.md]: Just to confirm, these are indeed versioned releases for the Protocol npm package. This sort of change doesn't need to be reflected in the changelog.

**Pull request merged**